android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistry.kt

```diff
@@ -491,24 +491,52 @@ constructor(
     gatewayModeHeaderValue: String? = null,
     searchPath: String,
     patientRelatedResourceTypes: MutableList<ResourceType>,
+    currentPage: Int = 1,
   ) {
-    val resultBundle =
-      if (gatewayModeHeaderValue.isNullOrEmpty()) {
-        fhirResourceDataSource.getResource(searchPath)
-      } else
-        fhirResourceDataSource.getResourceWithGatewayModeHeader(
-          gatewayModeHeaderValue,
-          searchPath,
-        )
+    var currentPageUrl = "$searchPath&_page=$currentPage&_count=4"
+
+
+    // Fetch pages until there are no more next pages
+    while (currentPageUrl.isNotBlank()) {
```

Comment:
Instead of using a `while` loop, why don't we just read the response from **resultBundle** and then make the next request if it exists?

Reply:
The while loop helps me to fetch all the pages. The latter implementation only allows me to fetch the first page (instead of fetching all the total number of items). So even when I put the if (resultBundle.link("next")) it only returns the first page that was fetched. I am not sure if there is a url that can assist to fetch all the items in the gateway.

Reply:
I think `next` should always have the next page request